internal/proxy/add.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# add.sh adds new dependencies to the Go Cloud module proxy. This script
+# should be run from the root of the repository.
+
+set -o pipefail
+
+function log() {
+	echo "add.sh: $@" 1>&2
+}
+
+function die() {
+	log "$@"
+	exit 1
+}
+
+if ! [[ -d .git ]]; then die 'this script must be run from the root of the repository'; fi
```

Comment:
This doesn't have to be an error: you can use the relative position from [the script's source directory](https://stackoverflow.com/questions/59895/getting-the-source-directory-of-a-bash-script-from-within) to make any working directory be okay.

Reply:
makeproxy.sh is called by add.sh and makeproxy.sh also assumes it's being called from the root dir of the repo. I don't think it's worth it to make these scripts runnable from anywhere. Better to just keep them simple and fail fast.

Reply:
That seems unfortunate for `makeproxy.sh`. Sent #1031 to fix that.

My main objection here is that the presence of the `.git` directory isn't really sufficient to test the condition that you're checking for. The most robust mechanism I can think of is by using the trick I linked to, which allows you to guarantee the working directory you wish, which means that there isn't an error condition for the script's user.

Reply:
Done.